Sources/_TestDiscovery/SectionBounds.swift

```diff
@@ -289,22 +291,68 @@ private func _sectionBounds(_ kind: SectionBounds.Kind) -> some Sequence<Section
   }
   return HMODULE.all.lazy.compactMap { _findSection(named: sectionName, in: $0) }
 }
-#else
+
+#elseif !SWT_NO_DYNAMIC_LINKING
+// MARK: - Missing dynamic implementation
+
 /// The fallback implementation of ``SectionBounds/all(_:)`` for platforms that
 /// support dynamic linking.
 ///
 /// - Parameters:
 ///   - kind: Ignored.
 ///
 /// - Returns: The empty array.
-private func _sectionBounds(_ kind: SectionBounds.Kind) -> [SectionBounds] {
-  #warning("Platform-specific implementation missing: Runtime test discovery unavailable (dynamic)")
-  return []
+private func _sectionBounds(_ kind: SectionBounds.Kind) -> EmptyCollection<SectionBounds> {
+#warning("Platform-specific implementation missing: Runtime test discovery unavailable (dynamic)")
+  return EmptyCollection()
 }
-#endif
+
 #else
 // MARK: - Statically-linked implementation
 
+/// A type representing the upper or lower bound of a metadata section.
+///
+/// This type uses the experimental `@_rawLayout` attribute to ensure that
+/// instances have fixed addresses. Use the `..<` operator to get a buffer
+/// pointer from two instances of this type.
+///
+/// On platforms that use static linkage and have well-defined bounds symbols,
+/// those symbols are imported into Swift below using `@_silgen_name`, another
+/// experimental attribute.
+@_rawLayout(like: CChar) private struct _SectionBound: @unchecked Sendable, ~Copyable {
```

Comment:
I'd much rather not use this here and use `var` for the section begins and ends and label them `nonisolated(unsafe)` so that you can take a pointer to them.

Reply:
I had that in an earlier commit, but I've seen the compiler try to optimize away the stable address, which puts us in a bad place obviously.

Reply:
The optimizer cannot optimize away grabbing the pointer to an inout value.

Reply:
`withUnsafePointer(to: &_testContentSectionBegin)` will always be stable.